### ISSUE TITLE
Sustainability of Cordova maintenance

### ISSUE THREAD

**Issue by brodybits** [apache/cordova#163]
I would like to flag this as a critical issue.

We have been adding one maintainer after another but continue to suffer from a lack of resources so badly needed to keep up with bug reports and the rapidly changing mobile platform APIs.

I think we need to have some sponsorship of the most important maintenance tasks including:

- timely updates to keep up with the mobile platform APIs
- bug fixes
- documentation
- community support
- modernization and cleanup of the existing JavaScript

I think the root cause is the loss of backing from a couple major companies.

We have discussed some ideas on the mailing list including:

- CodeFund.io
- [GitHub sponsors](https://github.com/sponsors) (unfortunately it tells me to join a wait list)
- Tidelift (I tried https://tidelift.com/lifter/search/npm/cordova-lib but it shows me a "needs subscribers" message)

We also discussed Bountysource but it does not seem to be right for project *sustainability*.

I think it would be ideal if some of the biggest corporate users would be willing to give something back to help us improve the sustainability.

Unfortunately the Apache Foundation "sponsorship" program does not seem prepared to support our needs on updates and bug fixes: http://www.apache.org/foundation/sponsorship.html#FundsGo

I do understand and respect that we would likely need to coordinate this kind of an effort with the Apache Foundation.

I think this is especially bad in comparison to competitive frameworks such as Capacitor and React Native which do seem to have stronger corporate backing.

While the original goal was to "cease to exist", I think lack of sustainability is not the way we want to reach this result.

I would also like to vote +1 to pin this issue in first place.

P.S. I am adding a disclaimer that my description and comments in this issue are my own and do not necessarily reflect the views of any other Apache Cordova maintainers, Apache Cordova project members, Apache Cordova PMC members, Apache Cordova board members, or ASF (Apache Software Foundation).

**Comment by brodybits** [comment]
First step: clarify the existing situation in the footer apache/cordova-docs#1020

**Comment by LifeIsStrange** [comment]
Amazon has recently launched an opensource sponsorship program:
https://aws.amazon.com/fr/blogs/opensource/aws-promotional-credits-open-source-projects/

They could provide free servers for e.g CI.

**Comment by LifeIsStrange** [comment]
Historically, apache Cordova was owned by Adobe Inc.
Maybe they would be interested in helping their child?

**Comment by brodybits** [comment]
Correction: Apache Cordova was never owned by Adobe Inc.

Adobe Inc. acquired PhoneGap then donated the IPR to the Apache through an incubator project, with license changed from MIT to Apache 2.0.

My understanding is that Adobe is no longer interested in sponsoring time for maintenance.

Major competitor is now Ionic Capacitor, which is under the simplistic MIT license and does not require the app to wait for non-standardized "deviceready" event.

Major advantage of Cordova is very limited amount of JavaScript needed in the WebView. In contrast, Ionic Capacitor seems to come bundled with more JavaScript or TypeScript in its core.

In terms of servers, my understanding is that as an open-source project, it should be no problem for us to get the git, build, and testing servers we need for free. Developer time to keep up with bug fixes, platform updates, documentation, and GitHub support is badly needed.

**Comment by LifeIsStrange** [comment]
@brodybits
While capacitor is arguably better than Cordova, it has not yet a great support of all 3000+ Cordova native plugins.

**Comment by brodybits** [comment]
I think Capacitor took some major efforts to remain compatible with the most important Cordova plugins. If they missed any major plugins, I would be interested.

**Comment by LifeIsStrange** [comment]
@brodybits I am not aware of any incompatibility, I just doubt they are reached full compatibility and would love to be proved otherwise.
Anyway this is off topic.

**Comment by brodybits** [comment]
I am sure they would love a report if you find any major compatibility
issues.

>

**Comment by jcesarmobile** [comment]
This is from latest activity report:

> ## Github activity:
>  
>  Issue close rate of 63%:
>  - 268 Github Issues created in the last 3 months
>  - 170 Github Issues closed in the last 3 months
>  
>  PR close rate of 180%:
>  - 78 Github Pull Requests created in the last 3 months
>  - 141 Github Pull Requests closed/merged in the last 3 months

Looks like a lot of work is still done in Cordova repos

**Comment by jcesarmobile** [comment]
From latest activity report

> ## Github activity:
>  
>  Issue close rate of 71%:
>  - 334 Github Issues created in the last 3 months
>  - 236 Github Issues closed in the last 3 months
>  
>  PR close rate of 87%:
>  - 288 Github Pull Requests created in the last 3 months
>  - 250 Github Pull Requests closed/merged in the last 3 months

**Comment by brodybits** [comment]
I am now closing this issue with comments that are my own and do not necessarily reflect the views or opinions of other Apache Cordova members or ASF (Apache Software Foundation):

Issue close rate and PR close rate definitely look good.

Maintainers have definitely been working quite hard to keep the issues and maintenance under control.

Unfortunately I have to say that I do see a few areas where we could still use some improvement, such as:

- Completely remove and replace all use of UIWebView on iOS
- cordova-windows@7.0.0 did not work for a number of people including myself due to an old issue with WinJS (see cordova-windows issue number 341, which I am *not* linking from here). Bug was fixed but never released for about half a year. I took time out of my own schedule to make the new cordova-windows release, which is unfortunately DOA due to issues with updating 
- I think there are many, many, many issues on cordova-docs that have been piling up over the months. I think it would be good if someone would have a chance to do a more thorough review of the documentation for both accuracy and clarity.
